lib/websocket.go

```diff
@@ -20,7 +43,16 @@ type WebsocketHandler interface {
 // and serves to maintain the list of connections
 type wsHandler struct {
 	// Collect all websocket connections
-	conns []*websocket.Conn
+	conns         map[string]*WSConn
+	connsLock     sync.Mutex
+	tokenSource   key.Store
+	subscriptions map[string]string
+	subLock       sync.Mutex
+}
+
+type WSConn struct {
```

Comment:
terrible name

Not sure what to call this. It's here to help us associate profileIDs and with connIDs in the "other" direction. maybe just calling it a `Conn` is okay? or would that be too confusing with the other kinds of conns?